contracts/HealthPass721.sol

```diff
@@ -11,24 +11,59 @@ contract HealthPass721 is ERC721URIStorage, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter private _tokenIds;
 
+ //We can brainstorm about it
+    struct VaccineDetails {
+     string category;
+     string country;
+     string region;
+     string vaccineVendor;
+    }
+
+  mapping(address => VaccineDetails[]) vaccineDetailsMapping ;
+
+  event CertificateIssued(uint,VaccineDetails);
+  string baseURL;
     
 
-    constructor() ERC721("HealthPass", "HPASS") {
-        
+    constructor(string memory _baseURL) ERC721("HealthPass", "HPASS") {
+        baseURL=_baseURL;
+    }
+
+   //Can be extended based on requirement
+   modifier isUserEligibleForCertificate(address user) {
```

Comment:
This is interesting. My idea when I created this contract is, we will capture each shot.  Each country/event might have different meaning about what do they mean by fully vaccinated and based on the age too.

Reply:
I was just extending it for the app, its like our app takes the ownership of unique minting of each certificate, as of now its a simple, but we can extend it, if we want to have restrictions